src/goto-programs/convert_nondet.cpp

```diff
@@ -87,8 +88,9 @@ static goto_programt::targett insert_nondet_init_code(
     source_loc,
     true,
     allocation_typet::DYNAMIC,
-    !nullable,
+    nullable,
```

Comment:
I'm conscious about this change, and it is correct.

For coherence with function `object_factory`, I renamed the parameter `assume_non_null` to `allow_null`, therefore the removal of the negation.